src/Source/AbstractPropertySourceIterator.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Exporter\Source;
+
+use Doctrine\ORM\Internal\Hydration\IterableResult;
+use Symfony\Component\PropertyAccess\Exception\UnexpectedTypeException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+use Symfony\Component\PropertyAccess\PropertyPath;
+
+abstract class AbstractPropertySourceIterator implements SourceIteratorInterface
+{
+    private const DATE_PARTS = [
+        'y' => 'Y',
+        'm' => 'M',
+        'd' => 'D',
+    ];
+    private const TIME_PARTS = [
+        'h' => 'H',
+        'i' => 'M',
+        's' => 'S',
+    ];
+
+    /**
+     * @var IterableResult
+     */
+    protected $iterator;
+
+    /**
+     * @var PropertyAccessor
+     */
+    protected $propertyAccessor;
+
+    /**
+     * @var string default DateTime format
+     */
+    protected $dateTimeFormat;
+
+    /**
+     * @var array<string>
```

Comment:
IMO we should either use

```suggestion
     * @var string[]
     * @psalm-var list<string>
```

or 

```suggestion
     * @var list<string>
```

or

```suggestion
     * @var string[]
```

Reply:
I wanted to introduce an ExportField type, so I would change to `@var array<string|ExportField>`.
But I can use `string[]` atm if you prefer.